### PR TITLE
Kusto: Add more feature such as database ooperations, check cluster/d…

### PR DESCRIPTION
**Pull request by radennis**
…atabase name availability, activate, deactivate, event hub, list skus, list resource skus

This checklist is used to make sure that common issues in a pull request are addressed. This will expedite the process of getting your pull request merged and avoid extra work on your part to fix issues discovered during the review process.

### PR information
- [ ] The title of the PR is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For information on cleaning up the commits in your pull request, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).
- [ ] Except for special cases involving multiple contributors, the PR is started from a fork of the main repository, not a branch.
- [ ] If applicable, the PR references the bug/issue that it fixes.
- [ ] Swagger files are correctly named (e.g. the `api-version` in the path should match the `api-version` in the spec).

### Quality of Swagger
- [ ] I have read the [contribution guidelines](https://github.com/Azure/azure-rest-api-specs/blob/master/.github/CONTRIBUTING.md).
- [ ] My spec meets the review criteria:
  - [ ] The spec conforms to the [Swagger 2.0 specification](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/2.0.md).
  - [ ] The spec follows the guidelines described in the [Swagger checklist](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md) document.
  - [ ] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.

### CODE REVIEW

**Comment by annatisch** [comment]
Thanks @radennis - there are 8 ARM errors in the CI pass if you could please take a look.
Additionally there's a number of warnings that should be acted on to improve the SDK experience.
https://github.com/Azure/azure-rest-api-specs/pull/3595/checks?check_run_id=11051261

The model validator for the examples is also failing:
https://travis-ci.org/Azure/azure-rest-api-specs/jobs/412994026

Thanks!

File: specification/azure-kusto/resource-manager/Microsoft.Kusto/2017-09-07-privatepreview/kusto.json [hunk 1]
```diff
@@ -294,6 +370,180 @@
...
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "#/definitions/CloudError"
+            }
+          }
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Kusto/clusters/{clusterName}/checkNameAvailability": {
```
Comment (ravbhatnagar) [inline specification/azure-kusto/resource-manager/Microsoft.Kusto/2017-09-07-privatepreview/kusto.json]: why are there 2 check name availability APIs?

File: specification/azure-kusto/resource-manager/Microsoft.Kusto/2017-09-07-privatepreview/kusto.json [hunk 2]
```diff
@@ -294,6 +370,180 @@
...
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "#/definitions/CloudError"
+            }
+          }
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Kusto/clusters/{clusterName}/skus": {
```
Comment (ravbhatnagar) [inline specification/azure-kusto/resource-manager/Microsoft.Kusto/2017-09-07-privatepreview/kusto.json]: why do you need a skus api at RG scope? SKU api should be at subscription scope

Comment (radennis) [reply specification/azure-kusto/resource-manager/Microsoft.Kusto/2017-09-07-privatepreview/kusto.json]: As part of the autoscale onboarding with Bin and Andy Shen I was requested to implement the skus api both at the subscription scope and the resource group scope for the autoscale to work

**Comment by annatisch** [comment]
Ping @radennis - were you able to take a look at the ARM warnings and model validation errors?

**Comment by annatisch** [comment]
Closing due to inactivity.

**Comment by radennis** [comment]
@annatisch could you please reopen the pull request. 
I have a design review with Gaurav to get this approved.

**Comment by ravbhatnagar** [comment]
This has been reviewed in person. Signing off from ARM.

**Comment by annatisch** [comment]
@radennis - please correct the errors in the Linter and Model validator CI passes.

**Comment by liatbezalel** [comment]
I will publish a new iteration as soon as I get "Writer" permission to this repository.

**Comment by annatisch** [comment]
What is the status of the updates?
@radennis, did you give @liatbezalel write permission to your fork?

**Comment by liatbezalel** [comment]
@annatisch @ravbhatnagar I created [a new PR](https://github.com/Azure/azure-rest-api-specs/pull/3895) including our latest version. Please review the new one.

**Pull request closed without merging**